### ISSUE TITLE
Detect and Narrow Parameter Sets based on parameters

### ISSUE THREAD

**Issue by JustinGrote** [PowerShell/vscode-powershell#3880]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.

### Summary

New-AzADSPcredential has 9 parameter sets
![image](https://user-images.githubusercontent.com/15258962/158689815-c8cde6e8-5387-4561-a5f9-2d1621076ba6.png)

If I specify parameters that narrow the possible parameter sets, invalid overloads should no longer be shown. Further, the frame should update to show the most recent parameterset.

### Proposed Design

This should be doable, it happens in other languages e.g. Typescript.

**Comment by andschwa** [comment]
Does this work in PowerShell itself? Like is it doable through the SMA completion API?

**Comment by SeeminglyScience** [comment]
SMA doesn't show the syntax stuff that we do, but it does have an API that might make it possible (`StaticParameterBinder`). It's pretty heavy though.

I'd say this is "nice to have" but even if a PR was provided, we'd need to really examine performance impact.

**Comment by JustinGrote** [comment]
Right, just throwing this up here as a nice to have to make sure it's documented at least.

**Comment by SeeminglyScience** [comment]
> Right, just throwing this up here as a nice to have to make sure it's documented at least.

Oh yeah for sure. If nothing else it's good fuel to justify improving related engine APIs, so please keep 'em coming 😁
